amundsen_application/api/metadata/v0.py

```diff
@@ -634,3 +634,60 @@ def _get_dashboard_metadata(*, uri: str, index: int, source: str) -> None:
         message = 'Encountered exception: ' + str(e)
         logging.exception(message)
         return make_response(jsonify({'dashboard': {}, 'msg': message}), HTTPStatus.INTERNAL_SERVER_ERROR)
+
+
+@metadata_blueprint.route('/table/<path:table_key>/dashboards', methods=['GET'])
+def get_related_dashboard_metadata(table_key) -> Response:
+    """
+    Call metadata service endpoint to fetch related dashboard metadata
+    :return:
+    """
+    try:
+        url = f'{app.config["METADATASERVICE_BASE"]}{TABLE_ENDPOINT}/{table_key}/dashboard/'
+        results_dict = _get_related_dashboards_metadata(url=url)
+        return make_response(jsonify(results_dict), results_dict.get('status_code', HTTPStatus.INTERNAL_SERVER_ERROR))
+    except Exception as e:
+        message = 'Encountered exception: ' + str(e)
+        logging.exception(message)
+        return make_response(jsonify({'dashboards': [], 'msg': message}), HTTPStatus.INTERNAL_SERVER_ERROR)
+
+
+@action_logging
+def _get_related_dashboards_metadata(*, url: str) -> Dict[str, Any]:
+
+    results_dict = {
+        'dashboards': {},
+        'msg': '',
+    }
+
+    try:
+        response = request_metadata(url=url)
+    except ValueError as e:
+        # envoy client BadResponse is a subclass of ValueError
+        message = 'Encountered exception: ' + str(e)
+        results_dict['msg'] = message
+        results_dict['status_code'] = getattr(e, 'code', HTTPStatus.INTERNAL_SERVER_ERROR)
+        logging.exception(message)
+        return results_dict
+
+    status_code = response.status_code
+    results_dict['status_code'] = status_code
+
+    if status_code != HTTPStatus.OK:
+        message = 'Encountered error: Related Dashboard Metadata request failed'
+        results_dict['msg'] = message
+        logging.error(message)
+        return results_dict
+
+    try:
+        dashboard_data_raw: dict = response.json().get('dashboards', [])
+        results_dict['dashboards'] = [marshall_dashboard_partial(dashboard) for dashboard in dashboard_data_raw]
```

Comment:
I got a type error here:
`List comprehension has incompatible type List[Dict[Any, Any]]; expected List[str]`

Any idea how to fix it @ttannis ? Thanks!

Reply:
I know for one `dashboard_data_raw` is typed as a `dict` here but it should be a `List`. Default on line 659 should also be `[]`. However even with those changes I still get the error. Time-boxing 30min to see if I can figure it out.

Reply:
@Golodhros I am not sure why Mypy think that it's `List[str]` but it seems above line's type is incorrect -- it seems it should be list of something. I wonder if this is because of that.

Reply:
So I think what's messing it up is the initial declaration of `results_dict`. I think since we initialize `results_dict` with an empty list, mypy infers that it is a `List[str]` by default. Then, when we try to assign it to a list of dashboard objects here, it complains. 

@Golodhros I was able to get the error to go away by directly returning the following on this line:
```
dashboard_data_raw = response.json().get('dashboards', [])
return {
    'dashboards': [marshall_dashboard_partial(dashboard) for dashboard in dashboard_data_raw],
    'msg': 'Success'
}
```

Reply:
nice find!